### PR TITLE
CORDA-716 Retire withTestSerialization

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: node/src/integration-test/kotlin/net/corda/node/services/AttachmentLoadingTests.kt [hunk 1]
```diff
@@ -96,23 +100,25 @@ class AttachmentLoadingTests {
     }
 
     @Test
-    fun `test that attachments retrieved over the network are not used for code`() {
+    fun `test that attachments retrieved over the network are not used for code`() = withoutTestSerialization {
```
Comment (andr3ej) [inline node/src/integration-test/kotlin/net/corda/node/services/AttachmentLoadingTests.kt]: the driver methods should be in a separate class, but i don't have time for that right now

File: testing/test-utils/src/main/kotlin/net/corda/testing/SerializationTestHelpers.kt [hunk 2]
```diff
@@ -109,8 +90,3 @@ private fun createTestSerializationEnv(label: String) = object : SerializationEn
         KRYO_CHECKPOINT_CONTEXT) {
     override fun toString() = "testSerializationEnv($label)"
 }
-
-private const val AMQP_ENABLE_PROP_NAME = "net.corda.testing.amqp.enable"
-
-// TODO: Remove usages of this function when we fully switched to AMQP
-private fun isAmqpEnabled(): Boolean = java.lang.Boolean.getBoolean(AMQP_ENABLE_PROP_NAME)
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/SerializationTestHelpers.kt]: unrelated obsolete stuff

File: testing/test-utils/src/main/kotlin/net/corda/testing/internal/InternalSerializationTestHelpers.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import net.corda.core.serialization.internal._contextSerializationEnv
+import net.corda.core.serialization.internal._inheritableContextSerializationEnv
+import net.corda.testing.SerializationEnvironmentRule
+
+/**
+ * For example your test class uses [SerializationEnvironmentRule] but you want to turn it off for one method.
+ * Use sparingly, ideally a test class shouldn't mix serializers init mechanisms.
+ */
+fun <T> withoutTestSerialization(callable: () -> T): T {
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/internal/InternalSerializationTestHelpers.kt]: non-api, only still exists because some of our tests mix driver/MockNode test methods with unit test methods

File: samples/irs-demo/cordapp/src/test/kotlin/net/corda/irs/api/NodeInterestRatesTest.kt [hunk 4]
```diff
@@ -19,6 +19,7 @@ import net.corda.node.internal.configureDatabase
 import net.corda.nodeapi.internal.persistence.CordaPersistence
 import net.corda.nodeapi.internal.persistence.DatabaseConfig
 import net.corda.testing.*
+import net.corda.testing.internal.withoutTestSerialization
```
Comment (mikehearn) [inline samples/irs-demo/cordapp/src/test/kotlin/net/corda/irs/api/NodeInterestRatesTest.kt]: Can you add a TODO to this unit test saying that it should be split up between integration tests and unit tests? As that sounds like the issue here.

File: testing/test-utils/src/main/kotlin/net/corda/testing/internal/InternalSerializationTestHelpers.kt (hunk 3 above)
Comment (mikehearn) [reply testing/test-utils/src/main/kotlin/net/corda/testing/internal/InternalSerializationTestHelpers.kt]: Can you remark in the api doc that this is why it exists and that it's scheduled for destruction once we split the test types up?

Comment (andr3ej) [reply testing/test-utils/src/main/kotlin/net/corda/testing/internal/InternalSerializationTestHelpers.kt]: i've added a todo with a reference to a jira that has all the context

File: samples/irs-demo/cordapp/src/test/kotlin/net/corda/irs/api/NodeInterestRatesTest.kt (hunk 4 above)
Comment (andr3ej) [reply samples/irs-demo/cordapp/src/test/kotlin/net/corda/irs/api/NodeInterestRatesTest.kt]: done, on withoutTestSerialization for maximum context

**Comment by andr3ej** [comment]
i'm going to have a chat with devrel about documentation tomorrow
